src/mango/src/mango_cursor_view.erl

```diff
@@ -379,14 +395,22 @@ view_cb({row, Row}, #mrargs{extra = Options} = Acc) ->
             ok = rexi:stream2(ViewRow),
             set_mango_msg_timestamp();
         {Doc, _} ->
-            put(mango_docs_examined, get(mango_docs_examined) + 1),
+            mango_execution_stats:shard_incr_docs_examined(),
             couch_stats:increment_counter([mango, docs_examined]),
             Process(Doc)
     end,
     {ok, Acc};
-view_cb(complete, Acc) ->
+view_cb(complete, #mrargs{extra = Options} = Acc) ->
+    ShardStats = mango_execution_stats:shard_get_stats(),
+    case couch_util:get_value(execution_stats_map, Options, false) of
+        true ->
+            Stats = ShardStats;
+        false ->
+            DocsExamined = maps:get(docs_examined, ShardStats),
+            Stats = {docs_examined, DocsExamined}
+    end,
```

Comment:
We should do the assignment to the stats variable outside of the case statement, I believe the compiler generates an "assignment from nested block" warning or some such here as well.

```erlang
    Stats = case couch_util:get_value(execution_stats_map, Options, false) of
        true ->
            ShardStats;
        false ->
            DocsExamined = maps:get(docs_examined, ShardStats),
            {docs_examined, DocsExamined}
    end,
```

Reply:
You are right, and it is completely logical.  I did not see any warnings from the compiler related to that, though.